scripts/autolabel.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster, Damiano Testa
+-/
+import Lean.Elab.Command
+
+/-!
+# Automatic labelling of PRs
+
+This file contains the script to automatically assign a GitHub label to a PR.
+
+`AutoLabel.mathlibLabels` contains an assignment of github labels to folders inside
+the mathlib repository. The script uses `git diff` to determine which files
+a PR modifies and then finds all labels that should be added based on these changes.
+
+For the time being, the script only adds a label if it finds a single unique label
+that would apply. If multiple labels are found, nothing happens.
+-/
+
+open Lean System
+
+namespace AutoLabel
+
+/--
+A `Label` consists of the
+* The `label` field is the actual GitHub label name.
+* The `dirs` field is the array of all "root paths" such that a modification in a file contained
+  in one of these paths should be labelled with `label`.
+* The `exclusions` field is the array of all "root paths" that are excluded, among the
+  ones that start with the ones in `dirs`.
+  Any modifications to a file in an excluded path is ignored for the purposes of labelling.
+-/
+structure Label where
+  /-- The label name as it appears on github -/
+  label : String
+  /-- Array of paths which fall under this label. e.g. `"Mathlib" / "Algebra"`.
+
+  For a label of the form `t-set-theory` this defaults to `#["Mathlib" / "SetTheory"]`. -/
+  dirs : Array FilePath := if label.startsWith "t-" then
+      #["Mathlib" / ("".intercalate (label.splitOn "-" |>.drop 1 |>.map .capitalize))]
+    else #[]
+  /-- Array of paths which should be excluded.
+  Any modifications to a file in an excluded path are ignored for the purposes of labelling. -/
+  exclusions : Array FilePath := #[]
+  deriving BEq, Hashable
+
+/--
+Mathlib labels and their corresponding folders. Add new labels and folders here!
+-/
+def mathlibLabels : Array Label := #[
+  { label := "t-algebra",
+    dirs := #[
+      "Mathlib" / "Algebra",
+      "Mathlib" / "FieldTheory",
+      "Mathlib" / "RingTheory",
+      "Mathlib" / "GroupTheory",
+      "Mathlib" / "RepresentationTheory",
+      "Mathlib" / "LinearAlgebras"] },
```

Comment:
```suggestion
      "Mathlib" / "LinearAlgebra"] },
```

Reply:
How about adding a `pathExists` check for all the dirs?

Reply:
Sorry for the confusion, I was playing around, forgot to add the "WIP" label and went offline...

There is a check in place that all paths exist and the workflow fails otherwise!